File: GunDetectionApp/src/app/Models/location.ts
import { Guid } from 'guid-typescript';

export class Location {
  id: Guid;
  idUser: Guid;
  name: string;


  SubLocations: SubLocation[];
}
export class SubLocation {
  id: Guid;
  idLocation: Guid; 
  name: string;
}

File: GunDetection.Models/Location.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace GunDetection.Models
{
    public class Location
    {
        public Guid Id { get; set; }
        public Guid IdUser { get; set; }
        public string Name { get; set; }

        [ForeignKey("IdLocation")]
        public ICollection<SubLocation> SubLocations { get; set; }
        [ForeignKey("IdLocation")]
        public ICollection<Camera> Cameras { get; set; }
    }
}
